Practise_Problems/orderingSystem.js

```diff
@@ -1,114 +1,115 @@
+                  break;
+                }
+
+                //If the price is less than items within the store. Insert
+                //Split the array into 2 parts. Insert new item into the middle of the first part.
+                //Then merge them together
+                else {
+                  let firstHalf = store.slice(0, tracker);
+                  let secondHalf = store.slice(tracker);
+
+                  firstHalf.push([[item], price]);
+                  store = firstHalf.concat(secondHalf);
+                  break;
+                }
               }
 
-              //If the price is less than items within the store. Insert
-              //Split the array into 2 parts. Insert new item into the middle of the first part.
-              //Then merge them together
-              else {
-                let firstHalf = store.slice(0, tracker);
-                let secondHalf = store.slice(tracker);
-
-                firstHalf.push([[item], price]);
-                store = firstHalf.concat(secondHalf);
-                break;
-              }
+              break;
             }
-
-            break;
           }
         }
+
+        //Store is empty. First item ever
+        else {
+          store.push([item], price);
+        }
       }
 
-      //Store is empty. First item ever
+      //This is VIEW
       else {
-        store.push([item], price);
+        result.push(store[visited][0].sort());
+        visited += 1;
       }
     }
 
-    //This is VIEW
-    else {
-      result.push(store[visited][0].sort());
-      visited += 1;
-    }
-  }
-
-  console.log("-------------------COMMANDS-------------------");
-  console.log(commands);
-  console.log("-------------------STORE-------------------");
-  console.log(store);
-  console.log("-------------------RESULT VIEW-------------------");
-
-  //Showing whole result array (debug purpose)
-  return result;
-
-  //Showing the Nth cheapest item based on # of visits.
-  //return result[result.length-1];
-};
-
-let commands = [
-  ["INSERT", "Pizza", 8],
-  ["INSERT", "Milk", 4],
-  ["INSERT", "Gum", 1],
-  ["INSERT", "Coffee", 4],
-  ["VIEW", "-", "-"],
-  ["INSERT", "Snack", 5],
-  ["VIEW", "-", "-"],
-  ["VIEW", "-", "-"],
-];
-
-console.log(machine(commands));
+    console.log("-------------------COMMANDS-------------------");
+    console.log(commands);
+    console.log("-------------------STORE-------------------");
+    console.log(store);
+    console.log("-------------------RESULT VIEW-------------------");
+
+    //Showing whole result array (debug purpose)
+    return result;
+
+    //Showing the Nth cheapest item based on # of visits.
+    //return result[result.length-1];
+  };
+
+  let commands = [
+    ["INSERT", "Pizza", 8],
+    ["INSERT", "Milk", 4],
+    ["INSERT", "Gum", 1],
+    ["INSERT", "Coffee", 4],
+    ["VIEW", "-", "-"],
+    ["INSERT", "Snack", 5],
+    ["VIEW", "-", "-"],
+    ["VIEW", "-", "-"],
+  ];
```

Comment:
thought: did you test with a huge command list?   thats where we saw slowdowns in the interview